OpenRA.Platforms.Default/Sdl2Input.cs

```diff
@@ -161,7 +161,11 @@ public void PumpInput(IInputHandler inputHandler)
 								Platform.CurrentPlatform == PlatformType.Windows)
 								Game.Exit();
 							else
-								inputHandler.OnKeyInput(keyEvent);
+							{
+								// Only register initial key press
+								if (e.key.repeat == 0)
```

Comment:
The chat itself still can handle repeating keys it just seems an exception to the backspace input has been made, which i think is due to how the widgets handle keyboard shortcuts. - This means that individual widgets require an individual filter as per requirement not the low level platform code.

I'm think this could be done via adding the `e.key.repeat` value into the `KeyInput` datatype, this would then mean each widget would need to be refactored to handle whether or not it should respond to key repeat or should be a singular press.

So I agree with @pchote Its far to low level of an implementation unfortunately, and it should be the widgets that make the decision I think that would be the ideal solution. but it does require a bit of refactoring.

So back to the drawing board on this one, could we un-merge and i will work on a better solution.